app/main.py

```diff
@@ -1,3 +1,86 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+from app.shop import Shop
+from app.customer import Customer
+
+
+def shop_trip() -> None:
+    with open("app/config.json", "r") as f:
+        data = json.load(f)
+        for customer in data["customers"]:
+            cust = Customer(
+                name=customer["name"],
+                products=customer["product_cart"],
+                location=customer["location"],
+                money=customer["money"],
+                car=customer["car"],
+            )
+            print(f"{cust.name} has {cust.money} dollars")
+            closest = {}
+            shop_names = {}
+            cust_prod = []
+            prod_price = []
+            sh_prod_names = []
+            prod_count = []
+            sum_all = []
+            for shop in data["shops"]:
+                sh = Shop(
+                    name=shop["name"],
+                    location=shop["location"],
+                    products=shop["products"]
+                )
+                fuel = cust.car["fuel_consumption"]
+                prod_total = [
+                    a * b for a, b in zip
+                    (sh.prod_price(), cust.prod_count())
+                ]
+                prod_total = sum(prod_total)
+                fuel_total = round(
+                    cust.fuel_shop(fuel, cust.dist_shop(sh.location)), 2
+                )
+                expenses = prod_total + fuel_total
+                closest[sh.name] = expenses
+                shop_names[sh.name] = sh.products
```

Comment:
You can remove this code with needed to a separate method or function. I believe shop.py is good place for this but it’s up to you